.busted

```diff
@@ -0,0 +1,39 @@
+-- vim: ft=lua tw=80 sw=2 ts=2
+
+return {
+  _all = {
+    verbose = true,
+    lazy = true,
+    helper = "test/functional/preload.lua",
+    lpath = "build/?.lua;runtime/lua/?.lua;./?.lua",
+    output = "test.busted.outputHandlers.nvim",
+  },
+  ---
+  functional = { ROOT = { "test/functional" } },
+  ---
+  apiFunc = { ROOT = { "test/functional/api" } },
```

Comment:
This is an extra layer of configuration that gains nothing. And it moves knowledge out of cmake, which is the wrong direction, something that I've made clear numerous times before, e.g. https://github.com/neovim/neovim/pull/15708#pullrequestreview-758064590

Reply:
have you seen https://github.com/neovim/neovim/pull/18674?

Reply:
> something that I've made clear numerous times before, e.g. [#15708 (review)](https://github.com/neovim/neovim/pull/15708#pullrequestreview-758064590)

I'm trying to remove the dependency on `make`, but also cleanup the interface of `cmake`

> This is an extra layer of configuration that gains nothing. And it moves knowledge out of cmake, which is the wrong direction

Currently there's a lot of overlap between `cmake/RunTests.cmake` and `helpers.lua`, and then some options use cmake vars and some rely on env vars.

Using a busted configuration file will helps us track (and override) all variables in a centralized manner, see `--Xhelper OPTIONS`, without repeating the default options that are already set in `.busted` or the helper.

Reply:
I've created another PR for the docs, and leaving the busted config file to be discussed in #18674 where it's actually used.

Reply:
> have you seen https://github.com/neovim/neovim/pull/18674?

yes, and still see no reason to add this extra layer.

Reply:
> What dependency? Nothing is dependent on make

that part is regarding the docs 😄 

> yes, and still see no reason to add this extra layer.

I still don't get how this is a new layer.

I'm hoping to consolidate the split between `cmake/RunTests.cmake` and `helpers.lua`, where each of those can toggle a few configuration, but not all 

https://github.com/neovim/neovim/blob/e7b3fd8ad603ba064ce9457a8866430c436c3737/cmake/RunTests.cmake#L9-L21

vs

https://github.com/neovim/neovim/blob/e7b3fd8ad603ba064ce9457a8866430c436c3737/test/functional/helpers.lua#L31-L46

isn't better to make full use of busted which is already used and going anywhere, and reduce the instances where we're  shoe-horning CMake into configuring **both** the test-environment **and** the test-suit? that's what I'm trying to achieve here. 

worse-case is we learn the requirements to run busted for tests that don't require RPC, and try to configure them all in `helpers.lua`, or learn what the blocker is.

Reply:
> isn't it better to make full use of busted, which is already used and isn't going anywhere,

`helpers.lua` also isn't going anywhere. I've seen several of your PRs where they are a solution in search of a problem, and I don't want to spend more time on such PRs. So I am skeptical about other similar changes. "Making full use of busted" means nothing. You need to present a clear case, in one short sentence, about the actual problem that is being solved. Not "it would be nice".